radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/ParticleGroup.java

```diff
@@ -112,8 +134,18 @@ public final ParticleGroupBuilder addParticle(Particle particle, Spin spin) {
 			return this;
 		}
 
+		public final ParticleGroupBuilder addMetaData(String key, String value) {
+			Objects.requireNonNull(key, "key is required");
+			Objects.requireNonNull(value, "value is required");
+
+			this.metaData.put(key, value);
+
+			return this;
+		}
+
+
 		public ParticleGroup build() {
-			return new ParticleGroup(ImmutableList.copyOf(this.particles));
+			return new ParticleGroup(ImmutableList.copyOf(this.particles), metaData);
```

Comment:
Appreciate this is old code, but I'm not sure we need to copy the list here and again in the constructor.

Reply:
I noticed that, I should have fixed it right away